keras_nlp/samplers/sampler.py

```diff
@@ -145,10 +145,8 @@ def compute_probabilities(self, logits):
         This will always be done in full precision, regardless of dtype, and
         scale by `temperature`.
         """
-        logits_dtype = logits.dtype
         logits = ops.cast(logits, "float32")
-        probs = keras.activations.softmax(logits / self.temperature)
-        return ops.cast(probs, logits_dtype)
+        return keras.activations.softmax(logits / self.temperature)
```

Comment:
Ah, thanks for correcting this! We should never downcast the probs. I noticed this when working on Mistral and LLaMA but forgot to push the change.